src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -5,39 +5,68 @@ import {
   FunctionCall,
   generalizeType,
   getNodeType,
+  IntType,
+  Literal,
+  LiteralKind,
   TupleExpression,
   TypeNode,
 } from 'solc-typed-ast';
 import { printNode } from '../../utils/astPrinter';
 import { CairoType } from '../../utils/cairoTypeSystem';
-import { NotSupportedYetError } from '../../utils/errors';
 import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createArrayTypeName, createNumberLiteral } from '../../utils/nodeTemplates';
 import { notNull } from '../../utils/typeConstructs';
 import { mapRange, narrowBigIntSafe, typeNameFromTypeNode } from '../../utils/utils';
 import { uint256 } from '../../warplib/utils';
 import { add, locationIfComplexType, StringIndexedFuncGen } from '../base';
 
 /*
-  Converts [a,b,c] into WM0_arr(a,b,c), which allocates new space in warp_memory
+  Converts [a,b,c] and "abc" into WM0_arr(a,b,c), which allocates new space in warp_memory
   and assigns the given values into that space, returning the location of the
   start of the array
 */
 export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
-  gen(node: TupleExpression): FunctionCall {
+  stringGen(node: Literal): FunctionCall {
+    // Encode the literal to the uint-8 byte representation
+    assert(node.kind === LiteralKind.String);
+    const byteCode = [];
+    for (let i = 0; i < node.hexValue.length; i++) {
+      byteCode.push(node.hexValue.charCodeAt(i));
+    }
```

Comment:
If the intention is to decode the utf8 into bytes I think this is not correct because `charCodeAt` does not return the number which represents the byte representation of the character. It returns instead a number  which  represents the range where the char is in utf8 from which you can then get the bytes representation of the char. [Wikipedia](https://en.wikipedia.org/wiki/UTF-8) explains better than me.

A possible solution for this is using the `Buffer` class like:
```
const a = Buffer.from("some string")
const byteCode = a.toJSON().data
```